subprojects/docs/src/docs/userguide/jvm/toolchains.adoc

```diff
@@ -65,6 +73,21 @@ Three tools are available:
 * A `JavaLauncher` which is the tool used by the `JavaExec` or `Test` tasks
 * A `JavadocTool` which is the tool used by the `Javadoc` task
 
+=== Integration with tasks relying on a Java tool or Java home
+
+Any tasks that can be configured with a Java tool path, or a Java home location, can benefit from toolchains.
+
+While you will not be able to wire a toolchain tool directly, they all have metadata that gives access to their full path or to the path of the Java installation they belong to.
+
+For example, you can configure the executable for a Kotlin compile tasks as follows:
+
+====
+include::sample[dir="snippets/java/toolchain-kotlin/groovy/",files="build.gradle[tags=compiler-kotlin]"]
+include::sample[dir="snippets/java/toolchain-kotlin/kotlin/",files="build.gradle.kts[tags=compiler-kotlin]"]
```

Comment:
Should we reference the kotlin feature request and say that we used kotlin just as a common showcase?

Reply:
Might be worth to point people at the feature request to nudge JB :)

Reply:
I would not link to a feature request from docs. However it is true that this example may have to change in the future if they adopt toolchains.
Issue is that I can't really come up with another example that has meaning and direct application.

Any idea?

Reply:
I think right now, the example is great as it is something people need to workaround currently and they and copy&paste your solution. Once kotlin is fixed, we can make up some example (or use a widely used community plugin that hasn't adopted toolchains yet).

For groovy, I tracked on the impl issue that we remove these doc sections again. Maybe we should have an issue with the kotlin work as well so we don't forget to change the docs